### PR TITLE
Payment link and capture fixes

### PR DESCRIPTION
**Pull request by BushraAsif**

### CODE REVIEW

**Review by shahbaz-mehar — COMMENTED** [review]
Changes have been reviewed.

File: altapay.php [hunk 1]
```diff
@@ -2879,45 +2879,46 @@ public function hookPaymentOptions($params)
...
+
+            if (version_compare(_PS_VERSION_, '1.7.0.0', '>=')) {
+                $cart = $this->context->cart;
+                $amountPaid = $cart->getOrderTotal(true, Cart::BOTH);
+                $currency = new Currency($cart->id_currency);
+                // Default values for Apple Pay label and supported networks
+                $apple_pay_label = 'Apple Pay';
+                $applepay_supported_networks = ['visa', 'masterCard', 'amex'];
+
+                $paymentMethods = Altapay_Models_Terminal::getActiveTerminalsForCurrency($currency->iso_code, (int) $this->context->shop->id);
```
Comment (emicha) [inline altapay.php:2895]: You are casting a string into int. Please understand that an empty string will be casted as 0, which is a shop that might exists.

image

File: altapay.php [hunk 2]
```diff
@@ -3041,11 +3042,22 @@ public function createTransaction($savecard,
...
         }
+
+        $languageId = $this->context->language->id;
+        $languageCode = $this->context->language->iso_code;
+        $shopId = $this->context->shop->id;
+
+        if ($parent_order) {
+            $shopId = $parent_order->id_shop;
+            $languageId = $parent_order->getCustomer()->id_lang;
+            $languageCode = Db::getInstance()->getValue('SELECT iso_code FROM ' . _DB_PREFIX_ . 'lang WHERE `id_lang` = ' . (int) $languageId);
```
Comment (emicha) [inline altapay.php:3053]: You are casting a string into int. Please understand that an empty string will be casted as 0, which is a language that might exists.

image

Comment (BushraAsif) [reply altapay.php]: Updated!

**Review by emicha — APPROVED** [review]

**Pull request merged**